From: Guoqing Jiang <guoqing.jiang@suse.com>
Date: Mon, 28 Feb 2022 08:59:39 +0800
Subject: [PATCH] kernel/livepatch: fix compile issue of core.c
Patch-mainline: Never, fixes out-of-tree driver
Modified-by-SEL: N/A 
References: bsn#110

We have ported the mainline patch to SEL-2.0

patches.mainline/v5.17-rc1-livepatch-Fix-kobject-refcount-bug-on-klp_init_patch.patch

which changed the return type of klp_init_patch_early, let's reflect the 
change here, otherwise.

kernel/livepatch/core.c: In function ‘klp_enable_patch’:
kernel/livepatch/core.c:1444:6: error: void value not ignored as it ought to be
 1444 | ret = klp_init_patch_early(patch);

Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 kernel/livepatch/core.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/kernel/livepatch/core.c b/kernel/livepatch/core.c
index 8d10e01bb3e0..f04891678cbf 100644
--- a/kernel/livepatch/core.c
+++ b/kernel/livepatch/core.c
@@ -1545,11 +1545,12 @@ int klp_enable_patch(struct klp_patch *patch)
 		return -EINVAL;
 	}
 	
-	ret = klp_init_patch_early(patch);
-	if (ret) {
+	if (!try_module_get(patch->mod)) {
 		mutex_unlock(&klp_mutex);
-		return ret;
+		return -ENODEV;
 	}
+
+	klp_init_patch_early(patch);
 	
 	ret = klp_init_patch(patch);
 	if (ret)
-- 
2.26.2

